### PR TITLE
Adds RETURN_IF_ERROR and ASSIGN_OR_RETURN macros.

### PR DESCRIPTION
**Pull request by tcsnfkx**
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/64)

### CODE REVIEW

**Review by efoxepstein — APPROVED** [review]

**Review by efoxepstein — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/64#-:-M_wDACw9X1Kbu-qknBP:bnfp4nl)* status: 0 of 1 files reviewed, all discussions resolved

**Review by efoxepstein — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/64)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @tcsnfkx)

File: src/main/cc/wfa/measurement/common/macros.h [hunk 1]
```diff
@@ -15,16 +15,44 @@
 #ifndef WFA_MEASUREMENT_COMMON_MACROS_H_
 #define WFA_MEASUREMENT_COMMON_MACROS_H_
 
+#ifndef RETURN_IF_ERROR
```
Comment (wangyaopw) [inline src/main/cc/wfa/measurement/common/macros.h:18]: why do you want to define a local copy here?
We are using the same macros from the PJC repo.

Comment (efoxepstein) [reply src/main/cc/wfa/measurement/common/macros.h:18]: Other C++ code shouldn't need a dependency on PJ&C for these macros. (And this code, along with other common code, should eventually be pulled to its own repo with minimal deps.)

**Pull request merged**